x-pack/plugin/identity-provider/src/test/java/org/elasticsearch/xpack/idp/saml/test/IdpSamlTestCase.java

```diff
@@ -22,30 +28,59 @@
 import org.opensaml.security.x509.X509Credential;
 import org.w3c.dom.Element;
 
-import javax.xml.XMLConstants;
 import javax.xml.transform.OutputKeys;
 import javax.xml.transform.Transformer;
-import javax.xml.transform.TransformerConfigurationException;
 import javax.xml.transform.TransformerException;
-import javax.xml.transform.TransformerFactory;
 import javax.xml.transform.dom.DOMSource;
 import javax.xml.transform.stream.StreamResult;
+import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.StringWriter;
 import java.io.UncheckedIOException;
 import java.io.Writer;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Path;
 import java.security.PrivateKey;
 import java.security.cert.CertificateException;
 import java.security.cert.X509Certificate;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
 
 import static org.opensaml.core.xml.config.XMLObjectProviderRegistrySupport.getUnmarshallerFactory;
 
 public abstract class IdpSamlTestCase extends ESTestCase {
 
+    private static Locale restoreLocale;
+
+    @BeforeClass
+    public static void localeChecks() throws Exception {
+        Logger logger = LogManager.getLogger(IdpSamlTestCase.class);
+        SamlInit.initialize();
+        if (isTurkishLocale()) {
+            // See: https://github.com/elastic/elasticsearch/issues/29824
```

Comment:
@tvernum your upstream issue seems to be resolved in opensaml 3.4 but we use 3.4.5 here and it still actually has the same problematic behavior, reproducible (without this fix) with 
```
./gradlew ':x-pack:plugin:identity-provider:test' --tests "org.elasticsearch.xpack.idp.saml.idp.SamlIdpMetadataBuilderTests.testMetadataGenerationWithAllParameters" -Dtests.seed=15A837259F3EDCCF -Dtests.security.manager=true -Dtests.locale=tr -Dtests.timezone=America/Glace_Bay -Dcompiler.java=13
```

which creates metadata with key usage : `<md:KeyDescriptor use="sıgnıng">` , note `ı` vs `i`

Reply:
They fixed the `KeyDescriptorUnmarshaller` but not the `KeyDescriptorMarshaller` 🤦‍♂